edb/edgeql/compiler/expr.py

```diff
@@ -985,10 +983,14 @@ def _infer_index_type(
     node_type = setgen.get_expr_type(expr, ctx=ctx)
     index_type = setgen.get_set_type(index, ctx=ctx)
 
-    str_t = env.schema.get('std::str', type=s_scalars.ScalarType)
-    bytes_t = env.schema.get('std::bytes', type=s_scalars.ScalarType)
-    int_t = env.schema.get('std::int64', type=s_scalars.ScalarType)
-    json_t = env.schema.get('std::json', type=s_scalars.ScalarType)
+    str_t = env.schema.get_by_name(sn.QualName('std', 'str'))
+    assert isinstance(str_t, s_scalars.ScalarType)
```

Comment:
I don't think this change is an improvement -- it's more verbose and produces worse messages on failure. (Though that won't happen in these cases, I suppose).

Though it looks like `get` is preserved, so this isn't necessary anyway?